activerecord/test/cases/adapters/mysql2/arel_table_alias_test.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+require "support/connection_helper"
+require "models/aircraft"
+
+module ActiveRecord
+  class MysqlArelTableAliasTest < ActiveRecord::Mysql2TestCase
+    test "it should create record with custom arel_table.table_alias" do
+      Aircraft.arel_table.table_alias = "a"
```

Comment:
Is there a way to trigger those failures without mutating internal and private state of the framework?

First, `arel_table` is private API of the framework. Nobody should be using it in their applications.
Second, if people are mutating an private attribute of the framework and expecting it to work, well, good luck with that.

Reply:
The more I look at this, the more I think this isn't an issue. While `Aircraft.arel_table.table_alias = "a"` works, mutating the relation object isn't guarantee to work. That is why `Arel::Table` has one `alias` method that return a new object instead of mutating the caller.

I think the right corse of action is just to remove the `table_alias=` writer from `Arel::Table`. It isn't used anywhere in the framework other than in a test that is asserting things unrelated to that attribute.

Although I'm ok with fixing visiting the AST when a TableAlias node is present in Delete, Update and Insert statements.

Reply:
Rafael, I'm really grateful for your feedback.  I learn something every time.  I just submitted a PR here that removes the `writer` method for `table_alias=`.   Let me know if you want me to do anything with the AST when a TableAlias node is present in Delete, Update and Insert statements.

Reply:
I'm hesitant about this. The ability of setting a default alias for a table seems quite useful to me.

I started on this journey because I wanted to reduce the size of generated sql queries because they were blowing up our observability tooling due to their size, especially when expanding the `SELECT *` fields.

If we say that we remove the ability of setting the alias on the model, what's the point of having such property there in the first place? Who would be setting it?

Reply:
This ability never existed. You can't use `Model.arel_table` in your applications. It isn't a documented feature. It is private API of the framework. It is there as an implementation detail, not as something that you can and should use in your application. `arel_table` is used to build the SQLs using the Active Record API. Nowhere we said you can call that method, mutate its internal values and expect that Active Record will just work for you.

If we had the ability to alias the table name of a model in all queries we would not do it telling users to call `arel_table`. It would be something like:

```
model User < ApplicationRecord
  self.table_alias = "u"
end
```

We don't have this feature, but I have nothing against implementing it.

This would be the right level to support such feature since we can make use the SQL being built will be correct. Mutating the state of AST nodes is really hard to make sure it will work, like we see in this PR that is reaching to all sorts of private API like instance variables that aren't supposed, or even worse, trying to see if a string contains a substring.

> Who would be setting it?
Nobody, that is why we are removing it. The framework is not setting it anywhere, and the framework is the only valid user.

> I started on this journey because I wanted to reduce the size of generated sql queries because they were blowing up our observability tooling due to their size, especially when expanding the `SELECT *` fields.

If you observability system can't accept a valid SQL query or your application, don't you agree that is the problem you should be solving? Sure you could alias the table to have one single character, but as soon your have a larger query that pass the tooling limit, even using aliases, your observability system will break again.

Reply:
Oh yeah, I agree that the problem that the SQL not being accepted is beyond the scope of rails, but given that it currently generates massive SQLs, especially when the table name is long, I think that it's a good functionality to have.

> This ability never existed. You can't use Model.arel_table in your applications. It isn't a documented feature.

I'm a bit perplexed by that statement, given examples such as https://apidock.com/rails/ActiveRecord/Core/ClassMethods/arel_table:
```rb
class Post < ActiveRecord::Base
  scope :published_and_commented, -> { published.and(arel_table[:comments_count].gt(0)) }
end
```

I now see that this isn't the official Rails documentation and that [the method is explicitly marked](https://github.com/rails/rails/blob/22fa76c1250a2847305b9325752d941dbaa55983/activerecord/lib/active_record/core.rb#L353-L354) as `:nodoc:`, but such examples exist abundantly a bit everywhere, so it's quite a surprise for me to learn that it's not part of the "public" API

Reply:
> https://apidock.com/rails/ActiveRecord/Core/ClassMethods/arel_table

I see. This is a common mistake. That isn't the official documentation. This one is https://api.rubyonrails.org/

> but such examples exist abundantly a bit everywhere, so it's quite a surprise for me to learn that it's not part of the "public" API

Yeah, no much we can do about that. People look at the implementation of Rails, find cool things, and blog about them, even though they aren't for public consumption.

Arel is very tricky. It is a cool concept and people are eager to play with it, what make content for it abundant. Not sure how to combat that.

Our desire is to make it public, but we didn't find yet an API that we are happy with, and releasing it with the current API means we need to commit to bad APIs like the one you found. Allowing to mutate an AST in the tree is a huge risk. A mutation could put the entire tree in an invalid state.